internal/backup/archiver_completeness_audited_test.go

```
@@ -0,0 +1,222 @@
+	}
+	return names
+}
+
+// newTestSocket creates a unix socket inside dir. A socket is a file type
+// tar.FileInfoHeader cannot represent ("sockets not supported"), which is how the
+// tests deterministically force a per-file archiving failure even when running as
+// root (permission-based failures do not block root). Returns the socket path.
+func newTestSocket(t *testing.T, dir string) string {
+	t.Helper()
+	sockPath := filepath.Join(dir, "x.sock")
+	l, err := net.Listen("unix", sockPath)
```

golangci-lint (2.12.2)

[error] net.Listen must not be called. use (*net.ListenConfig).Listen

(noctx)